model/histogram/float_histogram.go

```diff
@@ -982,3 +964,72 @@ func (i *allFloatBucketIterator) Next() bool {
 func (i *allFloatBucketIterator) At() Bucket[float64] {
 	return i.currBucket
 }
+
+func targetIdx(idx, schema, targetSchema int32) int32 {
+	return ((idx - 1) >> (schema - targetSchema)) + 1
+}
+
+func convertSpanToSchema(spans []Span, counts []float64, schema, targetSchema int32) ([]Span, []float64) {
+	targetSpans := make([]Span, 0)
+	targetCounts := make([]float64, 0)
+	// the index of bucket for origin schema
```

Comment:
Generally, we aim for easily readable English in comments throughout the Prometheus codebase. So:
- Start with a capital letter.
- End with a period.
- Don't leave out syntactically required words for brevity.

So this should read:
```Go
// The index of a bucket in the original schema.
```

This is probably still short enough to go on the same line, especially if using the `var` grouping suggested below:
```Go
var (
    // ...
    bucketIdx int32 // The index of a bucket in the original schema.
    // ...
)
```

Same applies to the other code comments below.